### PR TITLE
BL-737 Update Google analytics actions for navigational updates

### PR DESCRIPTION
**Pull request by cdoyle-temple**
Update GA actions (created in https://tulibdev.atlassian.net/browse/BL-551) for navigational updates from Fall 2018.

Add tracking for the following actions:

Action on menu items

Books/Articles/Journals/More headers
Books/Articles/Journals/More menu items
Actions on search results pages (ex. Clicking on links, buttons, etc.)

Make sure that Catalog Search actions still work for Books/Journals/More searches: Advanced Search links, Availability/Online buttons (don’t know if this is possible, since it can be a dropdown), Bookmark checkbox
Actions on record pages (ex. Clicking on links, buttons, etc.)

Breadcrumb links
Make sure that Catalog Search actions still work for Books/Journals/More searches: Back to Search button, Start Over button, Bookmark button, “Send To” dropdown/links, “Log in to see request options” link, Request button (only visible when logged in)

****

- Menu items replaced by Library Search link at top of pages

- Availability buttons and Bookmark checkboxes cannot be added due to their IDs being used for other purposes which require unique identifiers on each instance per page

- Added IDs to Books/Articles/Journals/More headers (modified render_nav_link method to accept id)
- Updated listeners.js for new navbar ids
- Updated breadcrumbs ids
- Added bookmarks button (trying the li wrapper as the link id is used for another purpose and too unique to be meaningful to analytics)
- Added Request Options link
- Added request button on record page to listeners.js
- Rubocop fixes

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 class BooksController < CatalogController
-  add_breadcrumb "Books", :back_to_books_path, only: [ :show ]
+  add_breadcrumb "Books", :back_to_books_path, options: { id: "breadcrumbs_book", only: [ :show ] }
```
Comment (dkinzer) [inline app/controllers/books_controller.rb]: @cdoyle-temple the `:only` field cannot be part of options:

```suggestion
  add_breadcrumb "Books", :back_to_books_path, { options: { id: "breadcrumbs_book" }, only: [ :show ] }
```

Comment (cdoyle-temple) [reply app/controllers/books_controller.rb]: I put them back into the hash and skipped rubocop before pushing.

File: app/controllers/journals_advanced_controller.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 class JournalsAdvancedController < AdvancedController
   copy_blacklight_config_from(JournalsController)
 
-  add_breadcrumb "Journals", :back_to_journals_path
+  add_breadcrumb "Journals", :back_to_journals_path, { options: { id: "breadcrumbs_journal" }, only: [ :show ] }
```
[the remark this answers is not in the document]
Comment (sensei100) [reply app/controllers/journals_advanced_controller.rb]: @dkinzer Why is :only a mistake?  The breadcrumbs are only supposed to be present on the record pages and advanced search pages.

[the remark this answers is not in the document]
Comment (dkinzer) [reply app/controllers/journals_advanced_controller.rb]: @sensei100 correct... this change is happening on `journal_advancec_controller.rb` which adds it to advanced page for journals.  Same is true for the other change but for articles.

[the remark this answers is not in the document]
Comment (sensei100) [reply app/controllers/journals_advanced_controller.rb]: @dkinzer That makes sense.  Thanks for clarifying.

**Comment by cdoyle-temple** [comment]
Woohoo!

**Pull request merged**